packages/core/src/components/tag-input/tagInput.tsx

```diff
@@ -15,6 +15,26 @@ import * as Utils from "../../common/utils";
 import { Icon, IconName } from "../icon/icon";
 import { ITagProps, Tag } from "../tag/tag";
 
+/**
+ * An enumeration of methods in which a `TagInput` value might have been added.
+ */
+export enum TagInputAddMethod {
```

Comment:
I would prefer the `const/type` approach so users can use string literals instead of importing this long-named type in order to `switch` on it.

Reply:
@giladgray Product teams have requested string-literal enums in the past, but happy to do a simpler type. Is this what you mean?

```tsx
type TagInputAddMethod = "default" | "blur" | "paste";
```

Or perhaps just inline it?

```tsx
onAdd?: (values: string[], method: "default" | "blur" | "paste") => boolean | void;
```

Reply:
I like the first one,

```
export type TagInputAddMethod = "default" | "blur" | "paste";
```

Reply:
Done